runtime/doc/msgpack_rpc.txt

```diff
@@ -445,5 +446,24 @@ states might be represented as separate modes.
 	curtab:	  Current Tabpage
 	tabs:	  List of Dicts [{ "tab": Tabpage, "name": String }, ...]
 
+							*ui-ext-cmdline*
+["cmdline_enter"]
+	Enter the cmdline.
+
+["cmdline_leave"]
+	Leave the cmdline.
+
+["cmdline_show", content, pos, firstc, prompt]
```

Comment:
Still, what about making `content` be a list of pairs `[highlight group, string]` (always one item `content=[["Normal", text]]` in this PR because cmdline highlighting PR is not finished yet)?